tools/compogen/pkg/gen/readme.go

```diff
@@ -255,12 +283,20 @@ func parseResourceProperties(o *objectSchema) []resourceProperty {
 	}
 
 	props := make([]resourceProperty, len(o.Properties))
-	for _, prop := range propMap {
-		// We can safely access the order pointer because it has been
-		// previously validated by the caller.
-		props[*prop.Order] = prop
+	idx := 0
+	for k := range propMap {
+		props[idx] = propMap[k]
+		idx += 1
 	}
 
+	// Note: The order might not be consecutive numbers.
```

Comment:
:ah: I was wondering about these changes

Reply:
We don't require the order of the properties to be consecutive from `0:len`. They might not be unique or consecutive.